app/models/order.rb

```diff
@@ -20,5 +21,25 @@ class Order < ApplicationRecord
   belongs_to :user
   belongs_to :course
 
+  store_accessor :progress, :total_lessons
+  store_accessor :progress, :project_complete
+  store_accessor :progress, :exam_complete
+  store_accessor :progress, :completed_lessons_ids
+
   validates :course_id, uniqueness: { scope: :user_id }
+
+  def lesson_complete?(lesson_id)
+    completed_lessons_ids.include?(lesson_id)
+  end
+
+  def percentage_count
+    completed_lessons_ids.count * 100 / total_lessons_count
+  end
+
+  def total_lessons_count
+    total = total_lessons
+    total += 1 unless exam_complete.nil?
+    total += 1 unless project_complete.nil?
```

Comment:
а почему бы не так сделать?
```suggestion
    total += 1 if exam_complete
    total += 1 if project_complete
```
так понятнее выглядит

Reply:
или можно progress.key?(:exam_complete)

Reply:
исправил